jcl/src/java.base/share/classes/com/ibm/tools/attach/target/Attachment.java

```diff
@@ -407,6 +413,14 @@ private static String startLocalAgent() throws IbmAttachOperationFailedException
 			throw new IbmAttachOperationFailedException("startLocalManagementAgent error starting agent:" + e.getClass() + " " + e.getMessage());		 //$NON-NLS-1$ //$NON-NLS-2$
 		}
 
+		try {
+			return saveLocalConnectorAddress();
+		} catch (IbmAttachOperationFailedException e) {
+			throw new IbmAttachOperationFailedException("startLocalManagementAgent: " + e.getMessage()); 
```

Comment:
This should set the original exception as the cause of the IbmAttachOperationFailedException